api/app/c_haines/severity_index.py

```diff
@@ -0,0 +1,453 @@
+
+
+def make_model_run_filename(
+        model: ModelEnum, level: str, date: str, model_run_start: str, forecast_hour: str):
+    """ Return the filename of the grib file.
+    The filename for each model differs slightly. """
+    if model == ModelEnum.GDPS:
+        return 'CMC_glb_{}_latlon.15x.15_{}{HH}_P{hhh}.grib2'.format(
+            level, date, HH=model_run_start, hhh=forecast_hour)
+    if model == ModelEnum.RDPS:
+        return 'CMC_reg_{}_ps10km_{}{HH}_P{hhh}.grib2'.format(
+            level, date, HH=model_run_start, hhh=forecast_hour)
+    if model == ModelEnum.HRDPS:
+        return 'CMC_hrdps_continental_{level}_ps2.5km_{date}{HH}_P{hhh}-00.grib2'.format(
+            level=level, date=date, HH=model_run_start, hhh=forecast_hour)
+    raise UnhandledPredictionModelType()
+
+
+def make_model_levels(model: ModelEnum):
+    """ Return list of layers. (The layers are named slightly differently for HRDPS) """
+    if model == ModelEnum.HRDPS:
+        return ['TMP_ISBL_0700', 'TMP_ISBL_0850', 'DEPR_ISBL_0850']
+    return ['TMP_ISBL_700', 'TMP_ISBL_850', 'DEPR_ISBL_850']
+
+
+def make_model_run_download_urls(model: ModelEnum,
+                                 now: datetime,
+                                 model_run_hour: int,
+                                 prediction_hour: int) -> Tuple[dict, datetime, datetime]:
+    """ Return urls to download model runs """
+
+    # hh: model run start, in UTC [00, 12]
+    # hhh: prediction hour [000, 003, 006, ..., 240]
+    levels: Final = make_model_levels(model)
+    # pylint: disable=invalid-name
+    hh = '{:02d}'.format(model_run_hour)
+    # For the global model, we have prediction at 3 hour intervals up to 240 hours.
+    hhh = format(prediction_hour, '03d')
+
+    base_url = make_model_run_base_url(model, hh, hhh)
+    date = get_file_date_part(now, model_run_hour)
+
+    adjusted_model_time = adjust_model_day(now, model_run_hour)
+    model_run_timestamp = datetime(year=adjusted_model_time.year,
+                                   month=adjusted_model_time.month,
+                                   day=adjusted_model_time.day,
+                                   hour=model_run_hour,
+                                   tzinfo=timezone.utc)
+
+    prediction_timestamp = get_prediction_date(model_run_timestamp, prediction_hour)
+    urls = {}
+    for level in levels:
+        filename = make_model_run_filename(model, level, date, hh, hhh)
+        urls[level] = base_url + filename
+
+    return urls, model_run_timestamp, prediction_timestamp
+
+
+class SourceInfo():
+    """ Handy class to store source information in . """
+
+    def __init__(self, projection, geotransform, rows: int, cols: int):
+        self.projection = projection
+        self.geotransform = geotransform
+        self.rows: int = rows
+        self.cols: int = cols
+
+
+def create_in_memory_band(data: numpy.ndarray, source_info: SourceInfo):
+    """ Create an in memory data band """
```

Comment:
This function documentation is a bit vague - could you describe what a data band is?